.github/workflows/tiledb-go.yml

```diff
@@ -65,6 +65,8 @@ jobs:
 
     # Tests TileDB-Go
     - name: Test TileDB-Go
+      env:
+        GOEXPERIMENT: cgocheck2
```

Comment:
Was this just preventative or did this produce some failures or output prior to these changes? I tried using this on `master` and the tests pass with no additional output, just curious if you had a different experience

Reply:
It's preventative, even though it might not matter in practice since we use `unsafe.Pointer` in lots of places, if I correctly understood what this switch does.